src/sdk/pynni/nni/compression/torch/utils/mask_conflict.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import logging
+import torch
+import numpy as np
+from .shape_dependency import ChannelDependency
+# logging.basicConfig(level = logging.DEBUG)
+_logger = logging.getLogger('FixMaskConflict')
+
+class MaskConflict:
+    def __init__(self, mask_file, model=None, dummy_input=None, graph=None):
+        """
+        MaskConflict fix the mask conflict between the layers that
+        has channel dependecy with each other.
+
+        Parameters
+        ----------
+        model : torch.nn.Module
+            model to fix the mask conflict
+        dummy_input : torch.Tensor
+            input example to trace the model
+        mask_file : str
+            the path of the original mask file
+        graph : torch._C.Graph
+            the traced graph of the target model, is this parameter is not None,
+            we donnot use the model and dummpy_input to get the trace graph.
+        """
+        # check if the parameters are valid
+        parameter_valid = False
+        if graph is not None:
+            parameter_valid = True
+        elif (model is not None) and (dummy_input is not None):
+            parameter_valid = True
+        if not parameter_valid:
+            raise Exception('The input parameters is invalid!')
+        self.model = model
+        self.dummy_input = dummy_input
+        self.graph = graph
+        self.mask_file = mask_file
+        self.masks = torch.load(self.mask_file)
+
+    def fix_mask_conflict(self):
+        """
+        Fix the mask conflict before the mask inference for the layers that
+        has shape dependencies. This function should be called before the
+        mask inference of the 'speedup' module.
```

Comment:
in some cases, if this function is not called, `speedup` will fail?

Reply:
Yes, the speedup module cannot handle mask conflict right now and will fail on the cases that have mask conflict(such as resnet, mobilenet, etc). Resolving conflicts before speedup will make it easier for speedup to support operations such as add and concat.

Reply:
then, maybe we should add resolve of mask conflict in this function: https://github.com/microsoft/nni/blob/master/src/sdk/pynni/nni/compression/torch/speedup/compressor.py#L160

Reply:
Yes, you are right, and I'm currently working on the support of the add&concat&group for the speedup module. I will submit a pr when I finish. Thanks~